include/psa/crypto_sizes.h

```diff
@@ -639,74 +639,149 @@
  */
-#define PSA_KEY_EXPORT_MAX_SIZE(key_type, key_bits)                     \
+#define PSA_EXPORT_KEY_OUTPUT_SIZE(key_type, key_bits)                      \
     (PSA_KEY_TYPE_IS_UNSTRUCTURED(key_type) ? PSA_BITS_TO_BYTES(key_bits) : \
-     (key_type) == PSA_KEY_TYPE_RSA_KEY_PAIR ? PSA_KEY_EXPORT_RSA_KEY_PAIR_MAX_SIZE(key_bits) : \
+     (key_type) == PSA_KEY_TYPE_RSA_KEY_PAIR ? PSA_KEY_EXPORT_RSA_KEY_PAIR_MAX_SIZE(key_bits) :     \
      (key_type) == PSA_KEY_TYPE_RSA_PUBLIC_KEY ? PSA_KEY_EXPORT_RSA_PUBLIC_KEY_MAX_SIZE(key_bits) : \
-     (key_type) == PSA_KEY_TYPE_DSA_KEY_PAIR ? PSA_KEY_EXPORT_DSA_KEY_PAIR_MAX_SIZE(key_bits) : \
+     (key_type) == PSA_KEY_TYPE_DSA_KEY_PAIR ? PSA_KEY_EXPORT_DSA_KEY_PAIR_MAX_SIZE(key_bits) :     \
      (key_type) == PSA_KEY_TYPE_DSA_PUBLIC_KEY ? PSA_KEY_EXPORT_DSA_PUBLIC_KEY_MAX_SIZE(key_bits) : \
-     PSA_KEY_TYPE_IS_ECC_KEY_PAIR(key_type) ? PSA_KEY_EXPORT_ECC_KEY_PAIR_MAX_SIZE(key_bits) : \
-     PSA_KEY_TYPE_IS_ECC_PUBLIC_KEY(key_type) ? PSA_KEY_EXPORT_ECC_PUBLIC_KEY_MAX_SIZE(key_bits) : \
+     PSA_KEY_TYPE_IS_ECC_KEY_PAIR(key_type) ? PSA_KEY_EXPORT_ECC_KEY_PAIR_MAX_SIZE(key_bits) :      \
+     PSA_KEY_TYPE_IS_ECC_PUBLIC_KEY(key_type) ? PSA_KEY_EXPORT_ECC_PUBLIC_KEY_MAX_SIZE(key_bits) :  \
+     0)
+
+/** The default IV size for a cipher algorithm, in bytes.
+ *
+ * The IV that is generated as part of a call to psa_cipher_encrypt() is always
+ * the default IV length for the algorithm.
+ *
+ * This macro can be used to allocate a buffer of sufficient size to store the
+ * IV output from psa_cipher_generate_iv() when using a multi-part cipher
+ * operation.
+ *
+ * See also #PSA_CIPHER_IV_MAX_SIZE.
+ *
+ * \param key_type      A symmetric key type that is compatible with algorithm
+ *                      alg.
+ * \param alg           A cipher algorithm (\c PSA_ALG_XXX value such that
+ *                      #PSA_ALG_IS_CIPHER(alg) is true).
+ *
+ * \return              The default IV size for the specified key type and
+ *                      algorithm. If the algorithm does not use an IV,
+ *                      return 0. If the key type or cipher algorithm
+ *                      is not recognized, or the parameters are incompatible,
+ *                      return 0. An implementation can return either 0
+ *                      or a correct size for a key type and cipher algorithm
+ *                      that it recognizes, but does not support.
+ */
+#define PSA_CIPHER_IV_LENGTH(key_type, alg)                             \
+    (PSA_ALG_IS_CIPHER(alg) && PSA_KEY_TYPE_IS_SYMMETRIC(key_type) ?    \
+     (PSA_BLOCK_CIPHER_BLOCK_LENGTH(key_type) > 1 ?                     \
+      PSA_BLOCK_CIPHER_BLOCK_LENGTH(key_type) :                         \
+      PSA_CIPHER_IV_MAX_SIZE) :                                         \
+     0)
+
+/** The maximum IV size for all supported cipher algorithms, in bytes.
+ */
+#define PSA_CIPHER_IV_MAX_SIZE      PSA_MAX_BLOCK_CIPHER_BLOCK_SIZE
```

Comment:
Same as above.

Reply:
Done

Reply:
This macro is no longer added by this PR (it was added in another PR in the meantime), so I'm marking this conversation as resolved.